### PR TITLE
Us560036 keytar installable using windows installer

### PR DESCRIPTION
**Pull request by uzuko01**
Adding config command group for the CLI, which currently includes 2 commands:

- set
- reset

### CODE REVIEW

File: packages/imperative/src/config/cmd/set/set.handler.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        break;
+      default :
+        throw new ImperativeError({
+          msg: "The setting you tried to change does not exist.",
+          additionalDetails: JSON.stringify(params)
+        });
+    }
+  }
+
+  private async overrideCredentialManager(value: string) {
```
Comment (AHumanFromCA) [inline packages/imperative/src/config/cmd/set/set.handler.ts]: Documentation comment needed for all methods

File: packages/imperative/__tests__/config/cmd/set/set.handler.test.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+jest.mock("../../../../../settings/src/AppSettings");
+
+import { CommandResponse, IHandlerParameters } from "../../../../../cmd";
```
Comment (AHumanFromCA) [inline packages/imperative/__tests__/config/cmd/set/set.handler.test.ts]: File should be 4 spaced

File: packages/cmd/__tests__/profiles/__snapshots__/CommandProfiles.test.ts.snap [hunk 3]
```diff
@@ -49,10 +49,6 @@ Object {
 }
 `;
 
-exports[`Command Profiles should detect a profile type mismatch 1`] = `"Expect Error: Command Profiles Internal Error: A profile supplied for type \\"banana\\" indicates a type of \\"strawberry\\""`;
```
[the remark this answers is not in the document]
Comment (AHumanFromCA) [reply packages/cmd/__tests__/profiles/__snapshots__/CommandProfiles.test.ts.snap]: Also probably not something directly related to this PR. Sorry about that

File: __tests__/__integration__/imperative/__tests__/__integration__/cli/config/cli.imperative-test-cli.config.set.credential-manager.integration.test.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            cliHomeEnvVar: "IMPERATIVE_TEST_CLI_CLI_HOME",
+            testName: "imperative_test_cli_test_config_set_credential_manager_command"
+        });
+    });
+
+    it("should override the default credential manager", () => {
+        const response = runCliScript(__dirname + "/__scripts__/set_credential_manager.sh",
+            TEST_ENVIRONMENT.workingDir, [TEST_CREDENTIAL_MANAGER]);
+        expect(response.status).toBe(0);
+        expect(response.stderr.toString()).toBe("");
```
Comment (AHumanFromCA) [inline __tests__/__integration__/imperative/__tests__/__integration__/cli/config/cli.imperative-test-cli.config.set.credential-manager.integration.test.ts]: Order should be:

```TypeScript
expect(response.stderr.toString()).toBe("");
expect(response.status).toBe(0);
```

File: __tests__/__integration__/imperative/__tests__/__integration__/cli/config/cli.imperative-test-cli.config.reset.credential-manager.integration.test.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        TEST_ENVIRONMENT = await SetupTestEnvironment.createTestEnv({
+            cliHomeEnvVar: "IMPERATIVE_TEST_CLI_CLI_HOME",
+            testName: "imperative_test_cli_test_config_reset_credential_manager_command"
+        });
+    });
+
+    it("should reset credential manager to the default value", () => {
+        const response = runCliScript(__dirname + "/__scripts__/reset_credential_manager.sh", TEST_ENVIRONMENT.workingDir);
+        expect(response.status).toBe(0);
+        expect(response.stderr.toString()).toBe("");
```
Comment (AHumanFromCA) [inline __tests__/__integration__/imperative/__tests__/__integration__/cli/config/cli.imperative-test-cli.config.reset.credential-manager.integration.test.ts]: Order should be:

```TypeScript
expect(response.stderr.toString()).toBe("");
expect(response.status).toBe(0);
```

**Comment by jelaplan** [comment]
I think you need to add examples, details on what can currently be configured, and what is currently configured. The config group is general purpose with a spec for set and reset commands. But, it doesn't expose anything to set or reset. I wonder what the value would be to include this in a release when customers wouldn't know what they can use it for. It resembles Easter egg ware. 

My understanding is that there will be an initial Object/Positional Argument for a credential manager. Given this, I think you should do the following
- List the credential manager object. For example when typing $ bright config set credentialmanager <name of manager>. I don't know the proper name of the object. If there is any concern that we are giving away the SCS commercial tech, then we shouldn't because it is perfectly fine to tell Zowe folks that Brightside has a credential manager setting. 
- Include examples of setting and resetting the credential manager. 
    - Assign the keytar Node module as the credential manager of record
        $ bright config set credentialmanager keytar

The lack of a list command is problematic. If a user were to set the credential manager or other config setting, there should be a way to list what they set. It doesn't make sense to put stuff in the box if we can't look into it to see what is there.

**Comment by jelaplan** [comment]
A few other suggestions
reset |  re     Reset a configuration setting to default or blank.
set     |  se    Set a configuration setting.

Reset a configuration setting to default or blank. 

Set a configuration setting.

**Comment by IgorCATech** [comment]
I agree with John here. Examples of how to set and reset a default credential manager would be appropriate - maybe somewhere below the "global options" section. Otherwise we will have to provide users with a procedure of how to set/reset credentials manager insider the "help-text" of the SCS plug-in.

**Comment by uzuko01** [comment]
spoke with  @jelaplan over the phone, agreed to add examples, aliases and change descriptions.
List command will be later on.

**Comment by IgorCATech** [comment]
@jelaplan Good point. I'm going to update our "command groups" doc.

**Comment by AHumanFromCA** [comment]
Updated Help text:

```
$ zd config reset --help

 COMMAND NAME
 ------------

   reset | re

 DESCRIPTION
 -----------

   Reset a configuration setting to default or blank.

 USAGE
 -----

   zowe config reset <configName> [options]

 POSITIONAL ARGUMENTS
 --------------------

   configName            (string)

      Setting name to reset

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text


 EXAMPLES
 --------

   - Reset the credential manager to default value:

      $ zowe config reset credential-manager
```

```
$ zd config set --help

 COMMAND NAME
 ------------

   set | se

 DESCRIPTION
 -----------

   Set a configuration setting

 USAGE
 -----

   zowe config set <configName> <configValue> [options]

 POSITIONAL ARGUMENTS
 --------------------

   configName            (string)

      Setting name

   configValue           (string)

      Value to set

 GLOBAL OPTIONS
 --------------

   --response-format-json  | --rfj (boolean)

      Produce JSON formatted data from a command

   --help  | -h (boolean)

      Display help text


 EXAMPLES
 --------

   - Set the default credential manager to @brightside/keytar:

      $ zowe config set credential-manager @brightside/keytar
```

**Comment by IgorCATech** [comment]
@AHumanFromCA this text looks much better. 
Also one question. Are there more settings (positional arguments) other than credential-manger to be added? If not, do you plan to add more of those?

**Comment by AHumanFromCA** [comment]
At this time there is only the credential manager, but in the future I could see other things get added to this group: (maybe an override for the help factory or logging levels)

For future reference, this PR partially completes this larger issue #126. So you could check that out to see some of the functionality that is still intended to be included in a future PR.

**Pull request merged**